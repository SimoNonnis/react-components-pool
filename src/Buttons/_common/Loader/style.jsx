import styled from 'styled-components';
import Spinner from '../../../Icons/Spinner';

export const SpinnerContainer = styled.div`
  position: relative;
`;

export const SpinnerStyled = styled(Spinner)`
  position: absolute;
  margin: 0 auto;
  left: 0;
  right: 0;
  top: -3px;
  ${({ size }) => size === 'large' && 'top: -5px'};
  ${({ size }) => size === 'small' && 'top: 0'};
`;
