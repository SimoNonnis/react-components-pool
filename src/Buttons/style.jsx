import React from 'react';
import styled, { css } from 'styled-components';
import { Link } from 'react-router-dom';
import {
  getColor,
  getColorOnHover,
  getBorderColor,
  getBorderColorOnHover,
  getBackgroundColor,
  getBackgroundColorOnHover,
  getPadding,
  getFontSize,
} from './utils';

const buttonStyle = css`
  -webkit-font-smoothing: antialiased;
  background-color: ${({ theme, styleType, type, disabled }) =>
    getBackgroundColor(theme, styleType, type, disabled)};
  border: 1px solid
    ${({ theme, styleType, type, disabled }) => getBorderColor(theme, styleType, type, disabled)};
  border-width: 1px;
  border-radius: ${({ rounded }) => `${rounded}px`};
  box-sizing: border-box;
  color: ${({ theme, styleType, type, disabled }) => getColor(theme, styleType, type, disabled)};
  cursor: ${({ disabled }) => (disabled ? 'no-drop' : 'pointer')};
  display: inline-block;
  font-family: ${({ theme }) => theme.base.fontFamily};
  font-size: ${({ size }) => getFontSize(size)};
  font-weight: 600;
  line-height: 24px;
  outline: 0;
  padding: ${({ size }) => getPadding(size)};
  text-align: center;
  text-shadow: none;
  white-space: nowrap;
  text-decoration: none;
  min-width: ${({ fullWidth, size }) => {
    if (fullWidth) return '100%';
    return size === 'small' ? '100px' : '140px';
  }};

  &:hover,
  &:focus {
    background-color: ${({ theme, styleType, type, disabled }) =>
      getBackgroundColorOnHover(theme, styleType, type, disabled)};
    border-color: ${({ theme, styleType, type, disabled }) =>
      getBorderColorOnHover(theme, styleType, type, disabled)};
    color: ${({ theme, styleType, type, disabled }) =>
      getColorOnHover(theme, styleType, type, disabled)};
    outline: 0;
  }
`;

const linkStyle = css`
  background-color: transparent;
  border: none;
  box-shadow: none;
  color: ${({ theme }) => theme.color.mediumBlue};
  text-decoration: underline;
  cursor: pointer;
  padding: 0;
  outline: 0;

  &:hover,
  &:focus {
    background-color: transparent;
    color: ${({ theme }) => theme.color.mediumBlue};
    text-decoration: none;
  }
`;

// eslint-disable-next-line
export const BtnLink = styled(({ external, fullWidth, to, theme, rounded, styleType, type, size, ...restProps}) => (external ? <a href={to} {...restProps} /> : <Link to={to} {...restProps} />))`
  ${buttonStyle};
`;
BtnLink.displayName = 'BtnLink';

export const Btn = styled.button`
  ${({ styleType, type }) => (styleType !== 'link' && type !== 'link' ? buttonStyle : linkStyle)};
`;
Btn.displayName = 'Btn';
