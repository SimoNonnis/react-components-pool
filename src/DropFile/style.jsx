import styled from 'styled-components';

import Dropzone from 'react-dropzone';

import UploadIcon from 'components/Icons/UploadIcon';
import Attention from 'components/Icons/Attention';
import P from 'components/Typography/P';
import Button from 'components/Buttons/Button';

export const StyledDropzone = styled(Dropzone)`
  border: 1px dashed ${({ theme }) => theme.color.mediumGrey};
  text-align: center;
  padding: ${({ theme }) => theme.spacing.lg} ${({ theme }) => theme.spacing.sm};
  cursor: pointer;
  background-color: ${({ theme }) => theme.color.white};
  width: 100%;
  box-sizing: border-box;
`;

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: ${({ theme }) => theme.color.veryDarkBlue};
  font-size: 16px;
  font-weight: 400;
`;

export const StyledUploadIcon = styled(UploadIcon)`
  color: ${({ theme }) => theme.color.lightGrey4};
`;

export const Label = styled(P)`
  margin-top: ${({ theme }) => theme.spacing.md};
`;

export const PickFileButton = styled(Button)`
  margin-top: ${({ theme }) => theme.spacing.sm};
`;

export const ContainerText = styled.div`
  padding: ${({ theme }) => theme.spacing.md};
  padding-bottom: 0;
  text-align: center;
`;

export const Message = styled(P)`
  color: ${({ error, theme }) => (error ? theme.color.red : theme.color.mediumGrey)};
  font-size: ${({ theme }) => theme.base.fontSize};
  font-style: ${({ error }) => (error ? 'normal' : 'italic')};
  margin-bottom: 0;
  margin-top: ${({ error, theme }) => (error ? theme.spacing.sm : '0')};
  ${({ error }) =>
    error &&
    `display: flex;
     align-items: center;
     justify-content: center;
  `};
`;

export const AttentionIcon = styled(Attention)`
  padding-right: ${({ theme }) => theme.spacing.xs};
  color: ${({ theme }) => theme.color.red};
`;
