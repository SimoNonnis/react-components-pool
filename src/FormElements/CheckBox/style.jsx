import styled from 'styled-components';

export const Container = styled.span`
  display: inline-block;
  position: relative;

  & + & {
    margin-left: 1em;
  }
`;
Container.displayName = 'Container';

export const Check = styled.input`
  opacity: 0;
  position: absolute;
  right: 0;

  &:checked + label:after {
    opacity: 1;
    transform: rotate(45deg) scale(1);
  }
`;
Check.displayName = 'Check';

export const Label = styled.label`
  cursor: pointer;
  display: inline-block;
  font-family: ${({ theme }) => theme.base.fontFamily};
  font-size: 16px;
  color: ${({ theme }) => theme.color.veryDarkBlue};
  ${({ theme, error }) => error && `color: ${theme.color.red}`};
  ${({ theme, disabled }) => disabled && `color: ${theme.color.lightGrey1}`};
  line-height: 24px;
  position: relative;
  user-select: none;
  padding-left: 30px;

  &:before {
    content: '';
    position: absolute;
    left: 0;
    top: 0;
    width: 20px;
    height: 20px;
    border: 1px solid;
    border-color: ${({ theme: { color }, disabled }) =>
      disabled ? color.lightGrey1 : color.stoneGrey};
    background-color: ${({ theme }) => theme.color.white};
    border-radius: 3px;
    box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.1);
  }

  ${({ theme, disabled }) =>
    disabled ? `&:hover { &:before { background-color: ${theme.color.lightGrey3}; } }` : null};
  &:after {
    border: solid ${({ theme }) => theme.color.mediumBlue};
    border-width: 0 3px 3px 0;
    content: '';
    font-family: ${({ theme }) => theme.base.fontFamily};
    font-size: 16px;
    height: 10px;
    left: 7px;
    top: 2.5px;
    opacity: 0;
    position: absolute;
    transform: rotate(45deg) scale(0);
    transition: all 0.2s;
    width: 5px;
  }
`;
Label.displayName = 'Label';

export const ExampleLabel = styled.div`
  color: ${({ theme }) => theme.color.green};
`;
