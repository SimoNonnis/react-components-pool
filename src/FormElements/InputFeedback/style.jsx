import styled from 'styled-components';

export const ErrorStyled = styled.div`
  color: ${({ theme }) => theme.color.red};
  font-size: ${({ theme }) => theme.base.fontSize};
  margin-top: 7px;
  min-height: 17px;
`;
