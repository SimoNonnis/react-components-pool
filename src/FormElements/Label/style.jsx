import styled from 'styled-components';

export const LabelStyled = styled.label`
  color: ${({ theme }) => theme.color.veryDarkBlue};
  font-family: ${({ theme }) => theme.base.fontFamily};
  ${({ theme, error }) => error && `color: ${theme.color.red}`};
  ${({ theme, disabled }) => disabled && `color: ${theme.color.lightGrey2}`};
  margin-bottom: 7px;
  display: block;
  font-size: ${({ theme }) => theme.base.fontSize};
  font-weight: 600;
`;
LabelStyled.displayName = 'LabelStyled';

export const Asterisk = styled.span`
  color: ${({ theme, disabled }) => (disabled ? theme.color.lightGrey2 : theme.color.red)};
  display: inline-block;
  margin-left: 3px;
`;
Asterisk.displayName = 'Asterisk';
