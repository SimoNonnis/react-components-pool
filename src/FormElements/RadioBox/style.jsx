import styled from 'styled-components';

export const Container = styled.span`
  display: inline-block;
  position: relative;

  & + & {
    margin-left: 1em;
  }
`;
Container.displayName = 'Container';

export const Radio = styled.input`
  opacity: 0;
  position: absolute;
  right: 0;

  &:checked + label:after {
    opacity: 1;
    transform: scale(1);
  }
`;
Radio.displayName = 'Radio';

export const Label = styled.label`
  cursor: pointer;
  font-family: ${({ theme }) => theme.base.fontFamily};
  color: ${({ theme }) => theme.color.veryDarkBlue};
  ${({ theme, error }) => error && `color: ${theme.color.red}`};
  ${({ theme, disabled }) => disabled && `color: ${theme.color.lightGrey1}`};
  font-size: 16px;
  line-height: 24px;
  position: relative;
  user-select: none;
  padding-left: 30px;
  display: inline-block;
  font-weight: 600;

  &:before {
    box-sizing: content-box;
    background-color: ${({ theme }) => theme.color.white};
    border-radius: 100%;
    border: 1px solid;
    border-color: ${({ theme: { color }, disabled }) =>
      disabled ? color.lightGrey1 : color.stoneGrey};
    box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.1);
    content: '';
    height: 20px;
    left: 0;
    position: absolute;
    top: 0;
    width: 20px;
  }

  ${({ theme, disabled }) =>
    disabled ? `&:hover { &:before { background-color: ${theme.color.lightGrey3}; } }` : null};
  &:after {
    box-sizing: content-box;
    background-color: ${({ theme }) => theme.color.mediumBlue};
    border-radius: 100%;
    content: '';
    opacity: 0;
    position: absolute;
    transform: scale(0);
    transition: all 0.2s;
    height: 12px;
    width: 12px;
    top: 5px;
    left: 5px;
  }
`;
Label.displayName = 'Label';
