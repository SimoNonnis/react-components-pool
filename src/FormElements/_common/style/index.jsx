import styled, { css } from 'styled-components';

const basicStyle = css`
  box-shadow: none;
  box-sizing: border-box;
  display: block;
  font-family: ${({ theme }) => theme.base.fontFamily};
  font-size: 16px;
  font-weight: inherit;
  height: auto;
  line-height: 24px;
  margin: 0;
  width: 100%;
  &:focus {
    outline: none;
    border-color: ${({ theme }) => theme.color.mediumBlue};
  }
`;

const borderStyle = css`
  border-radius: 2px;
  border: 1px solid;
`;

const defaultColor = css`
  ${borderStyle};
  border-color: ${({ theme }) => theme.color.stoneGrey};
  color: ${({ theme }) => theme.color.veryDarkBlue};
  caret-color: ${({ theme }) => theme.color.veryDarkBlue};
  &:focus {
    border-color: ${({ theme }) => theme.color.mediumBlue};
  }
  &::-webkit-datetime-edit-year-field,
  &::-webkit-datetime-edit-day-field,
  &::-webkit-datetime-edit-month-field {
    color: ${({ theme }) => theme.color.veryDarkBlue};
    font-weight: inherit;
  }
`;

const errorColor = css`
  ${borderStyle};
  border-color: ${({ theme }) => theme.color.red};
  &:focus {
    border-color: ${({ theme }) => theme.color.red};
  }
`;

const disabledColor = css`
  color: ${({ theme }) => theme.color.lightGrey2};
  border-color: ${({ theme }) => theme.color.lightGrey2};
  caret-color: ${({ theme }) => theme.color.lightGrey2};
  &:focus {
    border-color: ${({ theme }) => theme.color.lightGrey2};
  }
  &::-webkit-datetime-edit-year-field,
  &::-webkit-datetime-edit-day-field,
  &::-webkit-datetime-edit-month-field {
    color: ${({ theme }) => theme.color.lightGrey2};
    font-weight: inherit;
  }
`;

export const InputStyled = styled.input`
  ${basicStyle};
  ${defaultColor};
  ${({ error }) => error && errorColor};
  ${({ disabled }) => disabled && disabledColor};
  padding: 12px 15px;
`;

export const SelectStyled = styled.select`
  ${basicStyle};
  color: ${({ theme, disabled }) => (disabled ? theme.color.lightGrey2 : theme.color.veryDarkBlue)};
  padding: 13px 50px 13px 15px;
  -ms-appearance: none !important;
  appearance: none !important;
  background-image: none !important;
  background: transparent !important;
  border-radius: 0 !important;
  border: 0 !important;
  box-shadow: none !important;
  outline: 0 !important;
`;

export const SelectWrap = styled.div`
  ${defaultColor};
  ${({ error }) => error && errorColor};
  ${({ disabled }) => disabled && disabledColor};
  position: relative;
  &:after {
    border-left: 6px solid transparent;
    border-right: 6px solid transparent;
    border-top: 8px solid
      ${({ theme, disabled }) => (disabled ? theme.color.lightGrey2 : theme.color.veryDarkBlue)};
    content: '';
    height: 0;
    pointer-events: none;
    position: absolute;
    right: 20px;
    top: 50%;
    transform: translateY(-50%);
    transition: all 0.2s cubic-bezier(0.25, 0.46, 0.45, 0.94);
    width: 0;
    z-index: 2;
  }
`;

export const TextareaStyled = styled.textarea`
  ${basicStyle};
  ${defaultColor};
  ${({ error }) => error && errorColor};
  ${({ disabled }) => disabled && disabledColor};
  padding: 15px;
  resize: ${({ disabled }) => (disabled ? 'none' : 'vertical')};
`;
