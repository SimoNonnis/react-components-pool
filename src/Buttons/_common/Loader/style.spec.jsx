/* eslint-env jest */
import React from 'react';

import { SpinnerStyled } from './style';

describe('Test <SpinnerStyled />', () => {
  it('should render SpinnerStyled with empty size props', () => {
    const wrapper = shallow(<SpinnerStyled size="" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render SpinnerStyled with small size props', () => {
    const wrapper = shallow(<SpinnerStyled size="small" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render SpinnerStyled with large size props', () => {
    const wrapper = shallow(<SpinnerStyled size="large" />);
    expect(wrapper).toMatchSnapshot();
  });
});
