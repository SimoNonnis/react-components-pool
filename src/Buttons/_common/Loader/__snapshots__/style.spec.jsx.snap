// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <SpinnerStyled /> should render SpinnerStyled with empty size props 1`] = `
.c0 {
  position: absolute;
  margin: 0 auto;
  left: 0;
  right: 0;
  top: -3px;
}

<Spinner
  className="c0"
  fill="currentColor"
  height={40}
  size=""
  width={40}
/>
`;

exports[`Test <SpinnerStyled /> should render SpinnerStyled with large size props 1`] = `
.c0 {
  position: absolute;
  margin: 0 auto;
  left: 0;
  right: 0;
  top: -3px;
  top: -5px;
}

<Spinner
  className="c0"
  fill="currentColor"
  height={40}
  size="large"
  width={40}
/>
`;

exports[`Test <SpinnerStyled /> should render SpinnerStyled with small size props 1`] = `
.c0 {
  position: absolute;
  margin: 0 auto;
  left: 0;
  right: 0;
  top: -3px;
  top: 0;
}

<Spinner
  className="c0"
  fill="currentColor"
  height={40}
  size="small"
  width={40}
/>
`;
